### PR TITLE
massWindow fixes + solving bug in elreco, mutracking unc.

### PR DESCRIPTION
**Pull request by alesaggio**
This PR addresses the following items:
- Deletion of old files inherited from HH, not used in this analysis
- The function "isOutOfEllipse" has been removed. It suffices to have "isInEllipse(...)=False".
- The uncertainties on elreco and mutracking SFs were buggy. Fixed now.
- Added script to modify the rho histograms in order to add the bin-by-bin ttbarNorm uncertainty.

### CODE REVIEW

File: factories_ZA/basePlotter.py [hunk 1]
```diff
@@ -250,12 +249,12 @@ def generatePlots(self, categories, stage, requested_plots, weights, systematic=
...
-                    llIdIso_sf_dict["err_lep2_el"] = "({lep2}.isEl && (({lep2}.p4.Pt() < 20.) || ({lep2}.p4.Pt() > 80.)})) ? ({id}[{0}][0] * std::sqrt({reco}[{0}][{1}] * {reco}[{0}][{1}] + 0.01*{reco[{0}][{0}]} * 0.01*{reco[{0}][{0}]})) : ({id}[{0}][0] * {reco}[{0}][{1}])".format(self.lep2_fwkIdx, var_index, id=electron_id_branch, reco=electron_reco_branch, lep2=self.lep2_str)
+                    llIdIso_sf_dict["err_lep1_el"] = "(({lep1}.p4.Pt() < 20.) || ({lep1}.p4.Pt() > 80.)) ? ({id}[{0}][0] * std::sqrt({reco}[{0}][{1}] * {reco}[{0}][{1}] + 0.01*{reco}[{0}][{0}] * 0.01*{reco}[{0}][{0}])) : ({id}[{0}][0] * {reco}[{0}][{1}])".format(self.lep1_fwkIdx, var_index, id=electron_id_branch, reco=electron_reco_branch, lep1=self.lep1_str)
+                    llIdIso_sf_dict["err_lep2_el"] = "(({lep2}.p4.Pt() < 20.) || ({lep2}.p4.Pt() > 80.)) ? ({id}[{0}][0] * std::sqrt({reco}[{0}][{1}] * {reco}[{0}][{1}] + 0.01*{reco}[{0}][{0}] * 0.01*{reco}[{0}][{0}])) : ({id}[{0}][0] * {reco}[{0}][{1}])".format(self.lep2_fwkIdx, var_index, id=electron_id_branch, reco=electron_reco_branch, lep2=self.lep2_str)
```
Comment (OlivierBondu) [inline factories_ZA/basePlotter.py:253]: why remove the `isEl` ?

File: factories_ZA/basePlotter.py [hunk 2]
```diff
@@ -250,12 +249,12 @@ def generatePlots(self, categories, stage, requested_plots, weights, systematic=
...
-                    llIdIso_sf_dict["err_lep2_mu"] = "{lep2}.isMu ? (std::sqrt({tracking}[{0}][{1}] * {tracking}[{0}][{1}] + 0.005*{tracking}[{0}][{0}] * 0.005*{tracking}[{0}][{0}]) * {id}[{0}][0] * {iso}[{0}][0]) : ({tracking}[{0}][{1}] * {id}[{0}][0] * {iso}[{0}][0])".format(self.lep2_fwkIdx, var_index, tracking=muon_tracking_branch, id=muon_id_branch, iso=muon_iso_branch, lep2=self.lep2_str)
+                    llIdIso_sf_dict["err_lep1_mu"] = "(std::sqrt({tracking}[{0}][{1}] * {tracking}[{0}][{1}] + 0.005*{tracking}[{0}][{0}] * 0.005*{tracking}[{0}][{0}]) * {id}[{0}][0] * {iso}[{0}][0])".format(self.lep1_fwkIdx, var_index, tracking=muon_tracking_branch, id=muon_id_branch, iso=muon_iso_branch, lep1=self.lep1_str)
+                    llIdIso_sf_dict["err_lep2_mu"] = "(std::sqrt({tracking}[{0}][{1}] * {tracking}[{0}][{1}] + 0.005*{tracking}[{0}][{0}] * 0.005*{tracking}[{0}][{0}]) * {id}[{0}][0] * {iso}[{0}][0])".format(self.lep2_fwkIdx, var_index, tracking=muon_tracking_branch, id=muon_id_branch, iso=muon_iso_branch, lep2=self.lep2_str)
```
Comment (OlivierBondu) [inline factories_ZA/basePlotter.py:257]: same with `isMu` ?

**Review by OlivierBondu — CHANGES_REQUESTED** [review]
Thanks ! only a minor question !

File: factories_ZA/basePlotter.py (hunk 1 above)
Comment (alesaggio) [reply factories_ZA/basePlotter.py:253]: this check is not there for all the other SFs, not sure why I put it there. And the check on the flavor of the leptons is here already:
https://github.com/cp3-llbb/ZATools/blob/master/factories_ZA/basePlotter.py#L268#L277

Comment (OlivierBondu) [reply factories_ZA/basePlotter.py:253]: ok then, better to put every flavor check in the same place indeed :)

**Review by OlivierBondu — APPROVED** [review]

File: factories_ZA/basePlotter.py (hunk 1 above)
Comment (alesaggio) [reply factories_ZA/basePlotter.py:253]: Actually I'm thinking that the check should be there. The bug was the way the conditions are applied

Comment (alesaggio) [reply factories_ZA/basePlotter.py:253]: Ok, I remembered the bug. What the code was saying was:

```
if (lep1.isEl && (pT > ...)) {
    sf_uncertainty = unc1;
}
else {
    sf_uncertainty = unc2;
}
```

so in the else you're requiring the lepton to be a muon but with electron's SFs.
It should be instead:

```
if (lep1.isEl) {
    if ((pT > ...)) {
        sf_uncertainty = unc1;
    }
    else {
        sf_uncertainty = unc2;
    }
}
```

**Pull request merged**